src/screens/ReferFriends/ReferralContacts.js

```diff
@@ -0,0 +1,342 @@
+import debounce from 'lodash.debounce';
+import { connect } from 'react-redux';
+
+import { Wrapper } from 'components/Layout';
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import Button from 'components/Button';
+import SearchBlock from 'components/SearchBlock';
+import ListItemWithImage from 'components/ListItem/ListItemWithImage';
+import Checkbox from 'components/Checkbox';
+import EmptyStateParagraph from 'components/EmptyState/EmptyStateParagraph';
+import Spinner from 'components/Spinner';
+import { BaseText, TextLink } from 'components/Typography';
+import { Note } from 'components/Note';
+import Toast from 'components/Toast';
+
+import { fontStyles, spacing } from 'utils/variables';
+import {
+  getRemainingDailyInvitations,
+  isSameContact,
+  isSameContactData,
+} from 'utils/referrals';
+import { isValidPhone } from 'utils/validators';
+
+import type { NavigationScreenProp } from 'react-navigation';
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+import type { ReferralContact, SentInvitationsCount } from 'reducers/referralsReducer';
+
+import { setContactsForReferralAction } from 'actions/referralsActions';
+import { fetchPhoneContactsAction } from 'actions/phoneContactsActions';
+
+import { ADD_EDIT_USER, REFER_MAIN_SCREEN } from 'constants/navigationConstants';
+import { ALLOWED_DAILY_INVITES } from 'constants/referralsConstants';
+
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  setContactsForReferral: (contacts: ReferralContact[]) => void,
+  addedContactsToInvite: ReferralContact[],
+  phoneContacts: ReferralContact[],
+  isFetchingPhoneContacts: boolean,
+  isFetchingPhoneContactsComplete: boolean,
+  phoneContactsFetchError: boolean,
+  fetchPhoneContacts: () => void,
+  isEmailVerified: boolean,
+  isPhoneVerified: boolean,
+  alreadyInvitedContacts: ReferralContact[],
+  sentInvitationsCount: SentInvitationsCount,
+  userPhone: string,
+  userEmail: string,
+};
+
+type State = {
+  query: string,
+  selectedContacts: ReferralContact[],
+};
+const EmptyStateWrapper = styled.View`
+  width: 100%;
+  align-items: center;
+  margin: 20px 0 30px;
+`;
+
+const ButtonWrapper = styled.View`
+  justify-content: center;
+  padding: ${spacing.layoutSides}px;
+`;
+
+const StyledTextLink = styled(TextLink)`
+  ${fontStyles.regular};
+`;
+
+const MIN_QUERY_LENGTH = 3;
+
+const getInfoTypes = (isPhoneVerified, isEmailVerified) => {
+  // one or another should be true, otherwise feature is not available at all
+  if (!isPhoneVerified) {
+    return 'phone';
+  } else if (!isEmailVerified) {
+    return 'email';
+  }
+  return null;
+};
+
+const getFilteredContacts = (contacts, _query: string) => {
+  if (!_query || _query.length < MIN_QUERY_LENGTH) return contacts;
+  const query = _query.toUpperCase();
+  return contacts.filter(({ name, email = '', phone = '' }) => {
+    return name.toUpperCase().indexOf(query) > -1
```

Comment:
if name is null, it will crash on contacts search. Try to initiate it with { name = '', ... }